packages/dev/core/src/AudioV2/abstractAudio/streamingSoundInstance.ts

```diff
@@ -0,0 +1,34 @@
+import { Observable } from "../../Misc/observable";
+import { _AbstractSoundInstance } from "./abstractSoundInstance";
+import type { IStreamingSoundOptions, StreamingSound } from "./streamingSound";
+
+/** @internal */
+export abstract class _StreamingSoundInstance extends _AbstractSoundInstance {
+    private _resolvePreloadedPromise: () => void;
+
+    /** @internal */
+    public readonly onReadyObservable = new Observable<_StreamingSoundInstance>();
+
+    /** @internal */
+    public override options: IStreamingSoundOptions;
+
+    /** @internal */
+    public readonly preloadedPromise = new Promise<void>((resolve) => {
```

Comment:
A future note - this can be solved using `Promise.withResolvers` - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers . However, as this is not available for older browsers, this implementation is still safer.

Reply:
You can also use the `Deferred` class for this.